io/write_hdf5.cpp

```diff
@@ -46,7 +46,7 @@ namespace wash {
 
         double scalar_buffer[particle_count];
         int int_buffer[particle_count];
-        size_t dim = 2;
+        size_t dim = DIM;
```

Comment:
Can we just use `DIM` directly instead of copying into another variable? e.g. `for (size_t i = 0; i < DIM; i++) {` instead of `for (size_t i = 0; i < dim; i++) {`